### PR TITLE
Added BSIFieldCount diagnostics; refactored schema diagnostics

### PR DESCRIPTION
**Pull request by yuce**
## Overview

Implements #990 

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by yuce** [comment]
Sample payload:
```
{"BSIEnabled":true,"BSIFieldCount":2,"Cluster":"localhost:10101","ClusterID":"6be1a915-a8f8-4dc1-adfc-ad387975a472","GoRoutines":15,"Host":"localhost","LocalID":"6be1a915-a8f8-4dc1-adfc-ad387975a472","NumCPU":8,"NumFrames":3,"NumIndexes":3,"NumNodes":1,"NumSlices":3,"OpenFiles":30,"TimeQuantumEnaled":false,"Version":"v0.8.0-51-g0ad4272","uptime":1}
```

File: server.go [hunk 1]
```diff
@@ -723,3 +706,40 @@ type StatusHandler interface {
...
+				timeQuantumEnabled = true
+			}
+		}
+	}
+
+	d.Set("NumIndexes", numIndexes)
+	d.Set("NumFrames", numFrames)
+	d.Set("NumSlices", numSlices)
+	d.Set("BSIFieldCount", bsiFieldCount)
+	d.Set("BSIEnabled", bsiFieldCount > 0)
```
Comment (codysoyland) [inline server.go]: I think we can remove BSIEnabled since we now have BSIFieldCount.

File: server.go [hunk 2]
```diff
@@ -723,3 +706,40 @@ type StatusHandler interface {
...
+	d.Set("TimeQuantumEnaled", timeQuantumEnabled)
```
Comment (codysoyland) [inline server.go]: Check spelling: TimeQuantumEnaled -> TimeQuantumEnabled

**Comment by yuce** [comment]
@codysoyland updated.

**Comment by codysoyland** [comment]
Looks great!

**Comment by yuce** [comment]
Great!

**Pull request merged**